### ISSUE TITLE
PCC drop to 0.9601775503884877 in ttnn.gt after ttnn.softmax in N150

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-metal#39171]
### Describe the bug

- `ttnn.gt` produces PCC of `0.9601775503884877` vs CPU golden when applied to `ttnn.softmax` output near a threshold boundary (0.01). Intermediate op comparison shows softmax itself is accurate (PCC 0.9999), but the binary comparison amplifies small numerical differences at the decision boundary.
- **Pipeline:** softmax → slice → reshape → gt(threshold=0.01)
- **Input:** FLOAT32 tensor of shape [1, 8732, 91] (SSD classification logits)
- **Device:** N150
- SSD3300VGG16 model is affected by this drop - https://github.com/tenstorrent/tt-xla/issues/3570

| Operation         | TT vs CPU PCC |
|-------------------|---------------|
| Softmax           | 0.9999255355500597        |
| Slice+Reshape     | 0.9994476459903765        |
| **GT (final)**    | **0.9601775503884877**    |

## Questions

- @tenstorrent/metalium-developers-eltwise — Could you confirm that ttnn.gt only amplifies the error from the softmax output and is itself working correctly? Are there any known precision considerations when ttnn.gt operates on FLOAT32 inputs with dtype=BFLOAT16 output?

## Repro

```
git checkout kkannan/mar3_gt_pcc_drop_metal
git submodule update --init --recursive
git lfs pull --include '*.tensorbin'
# or you can unzip input_tensor.zip 

pytest tests/ttnn/unit_tests/operations/greater_than/test_exp1.py -svv
```

## Logs

- [mar5_exp1.log](https://github.com/user-attachments/files/25756844/mar5_exp1.log)
- [input_tensor.zip](https://github.com/user-attachments/files/25757036/input_tensor.zip)

**Comment by kamalrajkannan78** [comment]
> I'll take a look. Have you checked if numeric_stable and different levels of accuracy for compute kernel configs lead to better results?

thanks @bbradelTT . The attached sanity will reflect the below softmax config:

```
"ttnn.softmax"(%1) <{compute_config = #ttnn.device_compute_kernel_config<math_fidelity = hifi4, fp32_dest_acc_en = true>, dimension = 2 : si32, numericStable = true}> : (tensor<1x8732x91xf32, #ttnn_layout3>) -> tensor<1x8732x91xf32, #ttnn_layout3> 
```
I'm already using the highest accuracy settings — these match the softmax defaults in tt-metal when compute_kernel_config is not explicitly provided (None)

**Comment by bbradelTT** [comment]
@llongTT added you to take a look when you get the chance or re-assign

**Comment by ronnie00TT** [comment]
@kamalrajkannan78 @bbradelTT  I re-ran the test on the latest main branch, applying the same compute kernel config from the [comment above](https://github.com/tenstorrent/tt-metal/issues/39171#issuecomment-4005929529). And the log is

```
2026-07-02 15:10:26.452 | INFO     | tests.ttnn.unit_tests.operations.greater_than.test_exp1:test_gt_org:87 - Running CPU inference...
2026-07-02 15:10:26.479 | INFO     | tests.ttnn.unit_tests.operations.greater_than.test_exp1:test_gt_org:95 - === Intermediate Output Comparison (Softmax) ===
2026-07-02 15:10:26.487 | INFO     | tests.ttnn.unit_tests.operations.greater_than.test_exp1:test_gt_org:97 - Softmax Output PCC: 0.9999997362428968
2026-07-02 15:10:26.487 | INFO     | tests.ttnn.unit_tests.operations.greater_than.test_exp1:test_gt_org:100 - === Intermediate Output Comparison (Slice+Reshape) ===
2026-07-02 15:10:26.488 | INFO     | tests.ttnn.unit_tests.operations.greater_than.test_exp1:test_gt_org:102 - Slice+Reshape Output PCC: 0.9999970063364959
2026-07-02 15:10:26.488 | INFO     | tests.ttnn.unit_tests.operations.greater_than.test_exp1:test_gt_org:105 - === Final Output Comparison (Greater Than) ===
2026-07-02 15:10:26.488 | INFO     | tests.ttnn.unit_tests.operations.greater_than.test_exp1:test_gt_org:107 - GT Output PCC: 0.9975911159296115
```

I think this has been fixed in the latest main branch.
Does this look good to you?
CC @cmaryanTT

**Comment by bbradelTT** [comment]
@ronnie00TT that would be good.
@kamalrajkannan78 please take a look.

**Comment by kamalrajkannan78** [comment]
@ronnie00TT thanks for checking this 

I've tested out the sanity on top of [40afd08a3](https://github.com/tenstorrent/tt-metal/commit/40afd08a3e081994e4e768b053588a367dac813e) with [this](https://github.com/tenstorrent/tt-metal/issues/39171#issuecomment-4005929529) config. I am getting same PCC as yours. Hence closing the ticket

[jul7_exp1.log](https://github.com/user-attachments/files/29752362/jul7_exp1.log), [test_exp1.py](https://github.com/user-attachments/files/29752363/test_exp1.py)
